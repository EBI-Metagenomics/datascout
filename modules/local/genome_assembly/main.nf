process GENOME_ASSEMBLY {
    maxForks 1

    conda "./environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/d2/d2cc550ff67f8541d44dc2db1b5d2d2e1cfccfe8536222b49788deefde7460f0/data' :
        'community.wave.seqera.io/library/python_pip_biopython_requests:725bda83fb97ec48' }"
        
    debug true
    publishDir "${params.outdir}", mode: "copy"
    label "process_medium"

    tag "${meta}"

    errorStrategy  'retry'
    maxRetries 3

    input:
    tuple val(meta), path(genome_file)

    output:
    tuple val(meta), path ("*_reheaded_assembly.fasta"), emit: assembly_fa

    script:
    """
    if [[ "${genome_file}" == "" ]]; then
        genome_assembly.py --genome_id ${meta.id}
    else
        genome_assembly.py --genome_id ${meta.id} -f ${genome_file}
    fi

    if [[ ! -s "${meta.id}_reheaded_assembly.fasta" ]]; then
        rm -rf "${meta.id}_reheaded_assembly.fasta";
    fi
    """
}

// Download genome fasta file 