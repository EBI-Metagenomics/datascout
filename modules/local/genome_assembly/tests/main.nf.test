nextflow_process {

    name "Test Process GENOME_ASSEMBLY"
    script "modules/local/genome_assembly/main.nf"
    process "GENOME_ASSEMBLY"

    test("Should run without failures") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = tuple([ id:"GCA_003719455.1", ena_tax:"default"], [])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(
                process.out.assembly_fa
            ).match()
        }

    }

}
