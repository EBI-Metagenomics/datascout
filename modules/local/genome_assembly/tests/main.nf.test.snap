{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/1cfef12aadc1a01b62ee610061ea47ff/work/8b/6993f384f0d4c75ba08ead74749255/GCA_003719455.1_reheaded_assembly.fasta"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-08T15:15:19.076557"
    }
}